src/main/scala/sbtcrossproject/Platform.scala

```diff
@@ -0,0 +1,59 @@
+package sbtcrossproject
+
+/**
+ * Created by grinder on 17.07.16.
+ */
+abstract class Platform(val name: String, val path: String)
+
+object Shared extends Platform("shared", "shared/src/main") {
+
+  implicit class SharedTypeExt[T <: Shared.type](project: CrossProject[T]) {
+    def settings(settings: Seq[_root_.sbt.Def.Setting[_]]): CrossProject[T] =
+      project.settings(Shared, settings)
+
+    def shared                   = project.projects(Shared)
+    def sharedPath(path: String) = project.path(Shared, path)
+
+  }
+
+}
+
+object JVM extends Platform("jvm", "jvm/src/main") {
```

Comment:
You cannot bake `src/main`. What about test directories?

The "path" should be just `jvm`. This also means that `path` probably needs a better name.